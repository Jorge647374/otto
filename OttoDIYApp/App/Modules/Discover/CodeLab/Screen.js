import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  InstructionList,
  CodeLabNav,
  Modal } from 'App/Components'

import { Metrics } from 'App/Themes'

import Types from 'App/Services/PropTypes'

export default class Screen extends Component {
  static propTypes = {
    config: PropTypes.any,
    instructions: PropTypes.arrayOf(PropTypes.shape({
      ...Types.instruction
    })).isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    navHeight: PropTypes.number.isRequired,
    onConnect: PropTypes.func.isRequired,
    onSlidingComplete: PropTypes.func.isRequired,
    onChangeOrder: PropTypes.func.isRequired,
    onNavHeightChange: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    onNavPress: PropTypes.func.isRequired,
    onRun: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired
  }

  resolveSkills = () => {
    const { config } = this.props
    const { moves, skills } = config || {}

    const allSkills = []
    if (moves && moves.length > 0) {
      allSkills.push({
        id: 'moves',
        category: 'Moves',
        items: moves
      })
    }
    if (skills && skills.length > 0) {
      allSkills.push(...skills)
    }

    return allSkills
  }

  render () {
    const {
      instructions,
      showNotConnectedModal,
      navHeight,
      onConnect,
      onSlidingComplete,
      onChangeOrder,
      onNavHeightChange,
      onClose,
      onNavPress,
      onHideNotConnectedModal} = this.props

    const skills = this.resolveSkills()

    return (
      <Container>
        <InstructionList
          style={{marginBottom: navHeight + Metrics.unit}}
          instructions={instructions}
          onSlidingComplete={onSlidingComplete}
          onChangeOrder={onChangeOrder}
          onClose={onClose} />
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onConnect}
          template='NotConnected' />
        <CodeLabNav
          skills={skills}
          onPress={onNavPress}
          onNavHeightChange={onNavHeightChange} />
      </Container>
    )
  }
}
