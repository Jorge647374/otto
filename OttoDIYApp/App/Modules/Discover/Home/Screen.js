import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Card,
  List,
  FeaturedListItem,
  CompactListItem } from 'App/Components'

import { Metrics, Colors, Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <ScrollView style={{ marginBottom: 210 }}>
          <List title='Featured' cols={2}>
            <FeaturedListItem
              image={Images.featured.led}
              title='Blink'
              onPress={() => { onPress('Blink') }} />
            <FeaturedListItem
              icon={'heartbeat'}
              iconColor={Colors.red}
              title='Heart Beat'
              onPress={() => { onPress('Heart Beat') }} />
          </List>
          <List title='Projects'>
            <CompactListItem
              icon='circle-o'
              title='Hello World'
              onPress={() => { onPress('Hello World') }} />
            <CompactListItem
              icon='circle-o'
              title='Dance Show'
              onPress={() => { onPress('Dance Show') }} />
          </List>
        </ScrollView>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer style={{paddingTop: 0}}>
          <Card
            theme='light'
            title='DISCOVER'
            text={'Get inspired by what you can create. Build your own creations or remix others to jumpstart your ideas.'}
            button='Code Lab'
            textAlign='left'
            onPress={() => { onPress('Code Lab') }}
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
