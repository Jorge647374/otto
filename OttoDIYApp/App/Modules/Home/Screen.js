import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Container, List, ListItem, BottomNav } from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    onNavigatePress: PropTypes.func.isRequired
  }

  render () {
    const { onNavigatePress } = this.props
    return (
      <Container>
        <List>
          <ListItem
            title='Play & Explore'
            text='Let’s play'
            button='Play'
            onPress={() => { onNavigatePress('PlayerScreen') }} />
          <ListItem
            title='Discover'
            text='Let’s see what you can do'
            button='Discover'
            onPress={() => { onNavigatePress('DiscoverScreen') }} />
          <ListItem
            title='Learn'
            text='You’ll be a guru in no time'
            button='Learn'
            onPress={() => { onNavigatePress('LearnScreen') }} />
          <ListItem
            title='Lab'
            text='Get under the hood'
            button='Lab'
            onPress={() => { onNavigatePress('LabScreen') }} />
        </List>
        <BottomNav
          onRatePress={() => { onNavigatePress('RateScreen') }}
          onHomePress={() => {}}
          onSettingsPress={() => { onNavigatePress('SettingsScreen') }} />
      </Container>
    )
  }
}
