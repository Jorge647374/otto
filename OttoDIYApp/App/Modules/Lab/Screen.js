import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Links,
  Link,
  List,
  StatListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <List style={{ margin: Metrics.unit * 2 }} cols={2}>
          <StatListItem icon='bluetooth' title='bluetooth' value='Connected' />
          <StatListItem icon='battery' title='battery' value='85%' />
          <StatListItem icon='play' title='playtime' value='4.8h' />
          <StatListItem icon='code' title='remixes' value='5' />
          <StatListItem icon='book' title='lessons' value='2' />
          <StatListItem icon='gamepad' title='level' value='1' />
        </List>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer>
          <Links>
            <Link theme='light' text='Shop' onPress={onPress} />
            <Link theme='light' text='Customize' onPress={onPress} />
            <Link theme='light' text='Add a buddy' onPress={onPress} />
            <Link theme='light' text='View logs' onPress={onPress} />
          </Links>
        </Footer>
      </Container>
    )
  }
}
