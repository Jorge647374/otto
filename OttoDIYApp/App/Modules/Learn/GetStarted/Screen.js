import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card } from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    connected: PropTypes.bool.isRequired,
    onDone: PropTypes.func.isRequired,
    onConnectPress: PropTypes.func.isRequired
  }

  render () {
    const { connected, onDone, onConnectPress } = this.props

    const connectButton = (!connected)
      ? {title: 'Connect', onPress: onConnectPress}
      : {title: 'Next', onPress: null}

    const connectCard = (!connected)
      ? {title: 'Get connected', text: 'Letâ€™s get you connected & ready to start playing'}
      : {title: 'Nice work, you`re connected', text: null}

    return (
      <Container>
        <Carousel
          buttons={['Next', connectButton.title, 'Done']}
          onPress={[null, connectButton.onPress, null]}
          onDone={onDone}>
          <Card image={Images.hello} title='Hi there ðŸ‘‹' />
          <Card
            image={Images.bluetooth}
            title={connectCard.title}
            text={connectCard.text} />
          <Card
            image={Images.done}
            title='Woot woot ðŸŽ‰'
            text='Youâ€™re all setup & ready to play' />
        </Carousel>
      </Container>
    )
  }
}
