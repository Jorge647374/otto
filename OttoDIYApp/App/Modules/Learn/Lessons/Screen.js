import React, { Component } from 'react'
import { ScrollView } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  Footer,
  Modal,
  Card,
  List,
  ListHeader,
  CompactListItem } from 'App/Components'

import { Metrics } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    showNotReadyModal: PropTypes.bool.isRequired,
    onHideNotReadyModal: PropTypes.func.isRequired,
    onPress: PropTypes.func.isRequired
  }

  render () {
    const {
      showNotReadyModal,
      onHideNotReadyModal,
      onPress } = this.props
    return (
      <Container>
        <ScrollView style={{ marginBottom: 210 }}>
          <List>
            <ListHeader title='Lessons' />
            <CompactListItem
              icon='circle-o'
              title='Get Started'
              onPress={() => { onPress('Get Started') }} />
            <CompactListItem
              icon='circle-o'
              title='Play Time'
              onPress={() => { onPress('Play Time') }} />
            <CompactListItem
              icon='circle-o'
              title='Customize'
              onPress={() => { onPress('Customize') }} />
            <CompactListItem
              icon='circle-o'
              title='Code Remix'
              onPress={() => { onPress('Code Remix') }} />
          </List>
        </ScrollView>
        <Modal
          navigation={this.props.navigation}
          show={showNotReadyModal}
          onHidePress={onHideNotReadyModal}
          template='NotReady' />
        <Footer style={{paddingTop: 0}}>
          <Card
            theme='light'
            title='START YOUR NEXT LESSON'
            text={'Learn them skills!'}
            button='Start'
            textAlign='left'
            onPress={() => { onPress('Get Started') }}
            style={{textView: {marginBottom: Metrics.unit}}} />
        </Footer>
      </Container>
    )
  }
}
