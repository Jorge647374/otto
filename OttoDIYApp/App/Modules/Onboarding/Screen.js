import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card
} from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    onDone: PropTypes.func.isRequired
  }

  render () {
    const { onDone } = this.props

    return (
      <Container>
        <Carousel
          buttons={['Next', 'Next', 'Done']}
          onPress={[null, null, null]}
          onDone={onDone}>
          <Card
            image={Images.onboarding.one}
            title='Welcome to the Otto DIY App'
            text='Where you can learn coding and bring your robot to live'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.two}
            title='Otto awaits your commands'
            text='Otto DIY+, Eyes, Wheels & Humanoid supported out of the box!'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.three}
            title='Lots more to come! ðŸŽ‰'
            text='Games, more robots, customize your Otto and more fun!'
            style={{titleViewText: {fontSize: 22}}}
          />
        </Carousel>
      </Container>
    )
  }
}
