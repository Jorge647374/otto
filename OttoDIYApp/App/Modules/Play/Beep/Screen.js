import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Types from 'App/Services/PropTypes'

import {
  Container,
  List,
  SquareListItem,
  Modal} from 'App/Components'

export default class Screen extends Component {
  static propTypes = {
    sounds: PropTypes.arrayOf(Types.sound).isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    onPlay: PropTypes.func.isRequired,
    onHideNotConnectedModal: PropTypes.func.isRequired,
    onBack: PropTypes.func.isRequired
  }

  render () {
    const {
      sounds,
      showNotConnectedModal,
      onPlay,
      onHideNotConnectedModal,
      onBack
    } = this.props
    return (
      <Container>
        <List cols={2} scrollable>
          {sounds.map(sound => {
            return (
              <SquareListItem
                key={sound.key}
                icon='play'
                iconStyle={{marginLeft: 4}}
                title={sound.name}
                onPress={() => { onPlay(sound) }} />
            )
          })}
        </List>
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          onBack={onBack}
          template='NotConnected' />
      </Container>
    )
  }
}
