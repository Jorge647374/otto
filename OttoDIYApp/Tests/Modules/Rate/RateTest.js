import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Screen from 'App/Modules/Rate/Screen'

describe('Rate', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Screen onRatePress={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
