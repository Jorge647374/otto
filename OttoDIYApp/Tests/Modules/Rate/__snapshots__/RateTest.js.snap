// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate Component renders correctly 1`] = `
<Container>
  <Footer>
    <Card
      button="Rate"
      onPress={[Function]}
      styles={
        Object {
          "titleView": Object {
            "marginTop": 0,
          },
        }
      }
      text="Every rating means the world to us.

★★★★★"
      theme="light"
      title="LOVING THE APP?"
    />
  </Footer>
</Container>
`;
